modules/custom/domain_extention/domain_extention.module

```diff
@@ -0,0 +1,196 @@
+<?php
+
+/**
+ * @file Domain extension module.
+ */
+
+/**
+ * Default nodes count per page in content table.
+ */
+define('DOMAIN_EXTENTION_NODES_PER_PAGE', 50);
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Add fields that helps to implement VBO-style content selection feature with
+ * domain_content_admin form.
+ *
+ * @see domain_content_form().
+ */
+function domain_extention_form_domain_content_admin_alter(&$form, &$form_state, $form_id) {
+  global $user;
+
+  // Path to root folder of domain_extention module.
+  $path = drupal_get_path('module', 'domain_extention');
+
+  // Get filtered count of nodes that were selected by user before.
+  $filtered_count = variable_get("domain_extention_filtered_count_$user->uid", 0);
+
+  // In case filters haven`t been used yet - set $filtered_count to count of all
+  // nodes.
+  if (empty($filtered_count)) {
+    $filtered_count = _domain_extention_get_filtered_count();
+  }
+
+  // 0 - Do nothing
+  // 1 - Apply to default page nodes count (default = 50 nodes).
+  // 2 - Apply to all filtered nodes.
+  // @see domain_extention_admin_nodes_validate() for more info.
+  $form['select_nodes_on_page_or_all'] = array(
+    '#type' => 'hidden',
+    '#default_value' => '0',
+    '#weight' => 0,
+    '#attributes' => array(
+      'class' => array('select-nodes-on-page-or-all')
+    ),
+  );
+
+  $vars = array(
+    'selected_nodes_in_page' => t('Selected <strong>@default_nodes_count nodes</strong> in this page.', array('@default_nodes_count' => DOMAIN_EXTENTION_NODES_PER_PAGE)),
+    'select_all_nodes_in_table' => t('Select all @filtered_nodes_count nodes in this table.', array('@filtered_nodes_count' => $filtered_count)),
+    'selected_nodes_in_table' => t('Selected <strong>@filtered_nodes_count nodes</strong> in this table.', array('@filtered_nodes_count' => $filtered_count)),
+    'select_default_amount_nodes_in_page' => t('Select only @default_nodes_count nodes in this page.', array('@default_nodes_count' => DOMAIN_EXTENTION_NODES_PER_PAGE)),
+  );
+  $select_buttons_markup = theme('select_buttons', $vars);
+
+  $form['#attached']['css'][] =  "$path/css/domain_extention.css";
+  $form['#attached']['js'][] = "$path/js/domain_extention.js";
+  $form['#attached']['js'][] = array(
+    'data' => array(
+      'domainExtention' => array(
+        'filteredCount' => $filtered_count,
+        'selectButtonsMarkup' => $select_buttons_markup,
+        'defaultPageNodeCount' => DOMAIN_EXTENTION_NODES_PER_PAGE,
+      ),
+    ),
+    'type' => 'setting',
+  );
+
+  $form['#submit'][] = 'domain_extention_filter_form_submit';
+  $form['admin']['options']['submit']['#validate'][] = 'domain_extention_admin_nodes_validate';
+}
+
+/**
+ * Validate handler for domain_admin_nodes form. Determine whether to perform
+ * mass operation on all nodes or only on part of them.
+ */
+function domain_extention_admin_nodes_validate($form, &$form_state) {
+  $domain_info = domain_get_domain();
```

Comment:
What is the point in this variable out of usage scope?
Move it into the case where it is used.